File: 226-invert-binary-tree/226-invert-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
const invertTree = (node) => {
    if (!node) return null
    
    let temp = node.left
    node.left = node.right
    node.right = temp
    
    invertTree(node.left)
    invertTree(node.right)
    
    return node
}

File: 226-invert-binary-tree/226-invert-binary-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func invertTree(node *TreeNode) *TreeNode {
    if node == nil {
        return node
    }
    
    var tempNode = node.Left
    
    node.Left = node.Right
    node.Right = tempNode
    
    invertTree(node.Left)
    invertTree(node.Right)
    
    return node
}
